text/0032-minimal-relay.md

```diff
@@ -0,0 +1,190 @@
+rewards. Allowing DOT directly on the Staking parachain would simplify staking changes.
+
+Given the complexity, it would be pragmatic to include the Balances pallet in the Staking parachain
+in its first version. Any other systems that use overlapping locks, most notably governance, will
+need to recognise DOT held on both Asset Hub and the Staking parachain.
+
+There is more discussion about staking in a parachain in [Moving Staking off the Relay
+Chain](https://github.com/paritytech/polkadot-sdk/issues/491).
+
+#### Governance
+
+Migrating governance into a parachain will be less complicated than staking. Most of the primitives
+needed for the migration already exist. The Treasury supports spending assets on remote chains and
+collectives like the Polkadot Technical Fellowship already function in a parachain. That is, XCM
+already provides the ability to express system origins across chains.
+
+Therefore, actually moving the governance logic into a parachain will be simple. It can run in
+parallel with the Relay Chain's governance, which can be removed when the parachain has demonstrated
+sufficient functionality. It's possible that the Relay Chain maintain a Root-level emergency track
+for situations like [parachains
+halting](https://forum.polkadot.network/t/stalled-parachains-on-kusama-post-mortem/3998).
+
+The only complication arises from the fact that both Asset Hub and the Staking parachain will have
+DOT balances; therefore, the Governance chain will need to be able to credit users' voting power
+based on balances from both locations. This is not expected to be difficult to handle.
+
+## Drawbacks
+
+None at present.
+
+## Testing, Security, and Privacy
+
+Standard audit/review requirements apply. More powerful multi-chain integration test tools would be
+useful in developement.
+
+## Performance, Ergonomics, and Compatibility
+
+Describe the impact of the proposal on the exposed functionality of Polkadot.
+
+### Performance
+
+This is an optimization. The removal of public/user transactions on the Relay Chain ensures that its
+primary resources are allocated to system performance.
+
+### Ergonomics
+
+This proposal alters very little for coretime users (e.g. parachain developers). Application
+developers will need to interact with multiple chains, making ergonomic light client tools
+particularly important for application development.
+
+### Compatibility
+
+Implementing this proposal will require some changes to pallet APIs and/or a pub-sub protocol.
+Application developers will need to interact with multiple chains in the network.
+
+## Prior Art and References
+
+- [Transactionless Relay-chain](https://github.com/paritytech/polkadot/issues/323)
+- [Moving Staking off the Relay Chain](https://github.com/paritytech/polkadot-sdk/issues/491)
+
+## Unresolved Questions
+
+There remain some implementation questions, like how to use balances for both Staking and
```

Comment:
I don't think we need identity as its own chain. Then we have the question of how to determine if a feature should lives on its own parachain or coexisting with another parachain. e.g. we could have the identity pallet also lives on the governance parachain.

Reply:
The current migration paln results downtime. i.e. we have to freeze identity state, setup new genesis, run the new chain, and then it is up again. This is likely going to take weeks. Can we do better to reduce the downtime?

Reply:
As you've said, a lot of the use of Identity is for reading. While there is downtime, it is only to updating identities and providing judgements. The service still exists for rendering identity info. We can try to coordinate upgrades/merging PRs, but IMO the complexity of a zero-downtime migration outweighs the benefits.

Reply:
> I don't think we need identity as its own chain. Then we have the question of how to determine if a feature should lives on its own parachain or coexisting with another parachain.

Also, deposits are much lower on parachains. Current identity with 21 DOT is very expensive for a lot of people. Having it on a parachain makes it much more accessible.

Reply:
What if we use the current collectives chain as a hub for everything that's social/offchain related?